### PR TITLE
x86: use z_bss_zero and z_data_copy

### PR DESCRIPTION
**Pull request by dcpleung**
Instead of doing these in assembly, use the common `z_bss_zero()` and `z_data_copy()` C functions instead. This simplifies code a bit and we won't miss any additions to these two functions (if any) under x86 in the future (as x86_64 was actually not clearing gcov bss area). Also needs to move calling of `z_loapic_enable()` into `z_x86_prep_c()` as this needs to be after clearing of BSS.

### CODE REVIEW

File: arch/x86/core/ia32/crt0.S [hunk 1]
```diff
@@ -152,43 +152,6 @@ __csSet:
 	addl	$CONFIG_ISR_STACK_SIZE, %esp
 #endif
 
-#ifdef CONFIG_XIP
-	/*
```
Comment (andyross) [inline arch/x86/core/ia32/crt0.S:156]: Does this break XIP, or do we not support that with x86 anymore?  Worth a comment explaining at least.

Comment (dcpleung) [reply arch/x86/core/ia32/crt0.S:156]: We still have `qemu_x86_xip` to test XIP, and tests are still passing in CI. Note that the copying of data is still being done at the beginning or `z_x86_prep_c()` by calling `z_data_copy()`.

Comment (dcpleung) [reply arch/x86/core/ia32/crt0.S:156]: Found that it actually broke with `CONFIG_GDT_DYNAMIC=y` even though CI thinks everything is fine. Fixed it by moving things around.

**Review by andyross — APPROVED** [review]
Ah, cool.  Didn't even know we had a portable XIP layer

**Review by jhedberg — APPROVED** [review]

**Pull request merged**